clinica/utils/inputs.py

```diff
@@ -281,7 +282,7 @@ def find_sub_ses_pattern_path(
 ) -> None:
     """Appends the output path corresponding to subject, session and pattern in results.
 
-    If an error is encountered, its corresponding message is added to the list `error_encountered`.
+    If an error is encountered, its (subject,session) couple is added to the list `errors_encountered`.
```

Comment:
Actually I am wondering, is it possible (and wanted) to modify this function so that it returns a modified copy of the list is processes instead of the processed list instead ? The blocking point for me is that it is used for Parallel processing on a Manager().list() ; 

WDYT ?

Reply:
I think that it would be better indeed.
Currently, the parallel reader is implemented with a server process to handle the lists for results and errors.
I don't really see why we would need something like this for the task it is doing. It is both complicated and slow while the task is embarrassingly parallel (unless I missed something...).
However, I suggest we keep that for a dedicated PR such that we focus on this point then.
WDYT ?

Reply:
Sure, I will open a new issue for this !